Neos.Neos/Classes/Fusion/Helper/DimensionHelper.php

```diff
@@ -0,0 +1,202 @@
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Eel\ProtectedContextAwareInterface;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\ObjectManagement\DependencyInjection\DependencyProxy;
+
+/**
+ * Helper for nodes in various content dimensions.
+ */
+class DimensionHelper implements ProtectedContextAwareInterface
+{
+    /**
+     * @var ContentRepositoryRegistry
+     */
+    #[Flow\Inject(lazy: true)]
+    protected $contentRepositoryRegistry;
+
+    /**
+     * Get current dimension value for `node`.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.currentValue(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return string|null
+     */
+    public function currentValue(Node $node, string|ContentDimensionId $dimensionName): ?string
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        return $node->subgraphIdentity->dimensionSpacePoint->getCoordinate($contentDimensionId);
+    }
+
+    /**
+     * Get original dimension value for `node`. Differs form current value in cases of dimension fallback.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.originValue(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return string|null
+     */
+    public function originValue(Node $node, string|ContentDimensionId $dimensionName): ?string
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        return $node->originDimensionSpacePoint->getCoordinate($contentDimensionId);
+    }
+
+    /**
+     * Get default dimension value for the content repository defined by `node`.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findDefaultValue(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return string|null
+     */
+    public function findDefaultValue(Node $node, string|ContentDimensionId $dimensionName): ?string
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        $contentRepository = $this->contentRepositoryRegistry->get($node->subgraphIdentity->contentRepositoryId);
+        $rootValues = $contentRepository
+            ->getContentDimensionSource()
+            ->getDimension($contentDimensionId)
+            ?->getRootValues() ?: [];
+
+        $dimensionValue = reset($rootValues);
+
+        return $dimensionValue ? $dimensionValue->value : null;
+    }
+
+    /**
+     * Find all content dimensions in content repository defined by `node`.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findDimensions(node)
+     *
+     * @param Node $node
+     * @return array<string,ContentDimension>
+     */
+    public function findDimensions(Node $node): array
```

Comment:
hm, should we really pass in a Node here? I find this a bit weird, when we only read the CR ID from the node's subgraph identity..

Reply:
@skurfuerst as described above the node identifies the cr which may not be the default cr of the site. 

Suggestion as already suggested above `findDimensionsByNode` and `findDimensionsBySite`

Reply:
ToDo: Rename to `all` and accept `ContentRepositoryId | Node`